scripts/influxd-systemd-start.sh

```diff
@@ -1,5 +1,51 @@
 #!/bin/bash -e
 
+# Retrieve configuration value from influxd.
+function influxd_config() {
+    local header="${1}"
+    local target="${2}"
+
+    while IFS= read -r line
+    do
+        # The TOML specification allows for key-value pairs to be namespaced in
+        # collections. Therefore, it is not enough to grep for the desired key.
+        # This filters out all values that are outside the target section.
+        if [[ "${line}" =~ ^\[${header}\]$ ]]
+        then
+            local section_found=1
+        fi
+
+        # blank line signifies that the section has ended
+        if [[ ! ${line} ]]
+        then
+            local section_found=
+        fi
+
+        if [[ ${section_found:-} ]]
+        then
+                # Once within the target section, search the key-value pairs for the
+                # desired key. Currently, this only supports string values. Since
+                # this is used only for wal-dir and data-dir, this should be okay.
+                if [[ "${line}" =~ ^[[:space:]]*${target}[[:space:]]*=[[:space:]]\"(.*)\" ]]
+                then
+                    echo "${BASH_REMATCH[1]}" ; return
+                fi
+        fi
+    done <<< "$(influxd config -config /etc/influxdb/influxdb.conf ${INFLUXD_OPTS} 2>/dev/null)"
+}
+
+DATA_DIR="$( influxd_config data dir     )"
+WAL_DIR="$(  influxd_config data wal-dir )"
+if [[ ( -d "${DATA_DIR}" ) && ( -d "${WAL_DIR}" ) ]]
+then
+    # If this daemon is configured to run as root, influx_inspect hangs
+    # waiting for confirmation before executing. Supplying "yes" allows
+    # the service to continue without interruption.
+    yes | /usr/bin/influx_inspect buildtsi -compact-series-file \
+        -datadir "${DATA_DIR}"                                  \
+        -waldir  "${WAL_DIR}"
+fi
+
 /usr/bin/influxd -config /etc/influxdb/influxdb.conf $INFLUXD_OPTS &
```

Comment:
Still think you should unify config file location on lines 34 and 49 into a single parameter/constant.